src/imars3d/backend/diagnostics/rotation.py

```diff
@@ -25,40 +24,74 @@ def find_rotation_center(
     angles: array of angles in degrees or radians, which must match the order of arrays
     in_degrees: whether angles are in degrees or radians, default is True (degrees)
     atol_deg: tolerance for the search of 180 deg paris, default is 0.1 degrees
-    ncore: number of cores to use for parallel median filtering, default is -1, which means using all available cores.
+    max_workers: number of cores to use for parallel median filtering, default is 0, which means using all available cores.
 
     Returns
     -------
         rotation center in pixels
     """
-    # sanity check input
-    if arrays.ndim != 3:
-        raise ValueError("arrays must be 3D (valid radiograph stack)")
-    # locate 180 degree pairs
-    atol = atol_deg if in_degrees else np.radians(atol_deg)
-    idx_low, idx_hgh = find_180_deg_pairs_idx(angles, atol=atol, in_degrees=in_degrees)
-    # process
-    ncore = mproc.mp.cpu_count() if ncore == -1 else int(ncore)
-    # use shared memory model and tqdm wrapper for multiprocessing to reduce
-    # runtime memory footprint
-    with SharedMemoryManager() as smm:
-        # create the shared memory
-        shm = smm.SharedMemory(arrays.nbytes)
-        # create a numpy array point to the shared memory
-        shm_arrays = np.ndarray(
-            arrays.shape,
-            dtype=arrays.dtype,
-            buffer=shm.buf,
-        )
-        # copy data
-        np.copyto(shm_arrays, arrays)
-        # map the multiprocessing calls
-        rst = process_map(
-            find_center_pc,
-            [shm_arrays[il] for il in idx_low],
-            [shm_arrays[ih] for ih in idx_hgh],
-            max_workers=ncore,
-            desc="Finding rotation center",
+
+    arrays = param.Array(doc="3D array of images, the first dimension is the rotation angle omega.")
+    angles = param.Array(doc="array of angles in degrees or radians, which must match the order of arrays")
+    in_degrees = param.Boolean(default=True, doc="whether angles are in degrees or radians, default is True (degrees)")
+    atol_deg = param.Number(
+        default=1e-3,
+        doc="tolerance for the search of 180 deg paris, default is 0.1 degrees",
+    )
+    max_workers = param.Integer(
+        default=0,
+        bounds=(0, max(1, multiprocessing.cpu_count() - 2)),
```

Comment:
can you change the bounds here to
```
bounds=(0, None)
```

This is because the users might be running the ui from a different node in the future, so we should not limit the core numbers based on the ui node.